helpers/setup_json_generator.py

```
@@ -0,0 +1,297 @@
+
+        vpool_dict = {'backend_name': backend_name,
+                      'preset': preset,
+                      'storage_ip': storage_ip,
+                      'proxies':1,
+                      'fragment_cache': {
+                          'strategy': {'cache_on_read': False, 'cache_on_write': False},
+                          'location': "disk"
+                      },
+                      'block_cache': {
+                          'strategy': {'cache_on_read': False, 'cache_on_write': False},
+                          'location': "disk"
+                      },
+                      'storagedriver':
+                          {
+                              "sco_size": 4,
+                              "cluster_size": 4,
+                              "volume_write_buffer": 512,
+                              "strategy": "none",
+                              "global_write_buffer": 20,
+                              "global_read_buffer": 0,
+                              "deduplication": "non_dedupe",
+                              "dtl_transport": "tcp",
+                              "dtl_mode": "sync"
+                          }
+                      }
+        self.json['setup']['storagerouters'][storagerouter_ip]['vpools'][vpool_name] = vpool_dict
+
+    def _change_cache(self, storagerouter_ip, vpool, block_cache=True, fragment_cache=True, on_read=True, on_write=True):
+        try:
+            vpool = self.json['setup']['storagerouters'][storagerouter_ip]['vpools'][vpool]
+            if block_cache is True:
+                vpool['block_cache']['strategy']['cache_on_read'] = on_read
+                vpool['block_cache']['strategy']['cache_on_write'] = on_write
+            if fragment_cache is True:
+                vpool['fragment_cache']['strategy']['cache_on_read'] = on_read
+                vpool['fragment_cache']['strategy']['cache_on_write'] = on_write
+
+        except KeyError as e:
```

Kinda useless error message
The message of a `KeyError` is always the key which it couldn't find
So re-raising this with a `ValueError` is unnecessary, just let it raise the `KeyError` or show a more relevant error message